scripts/ci/ws12_suite_shards.py

```
@@ -0,0 +1,188 @@
+    with path.open("rb") as handle:
+        return tomllib.load(handle)
+
+
+def _required_text(record: dict, key: str, label: str) -> str:
+    value = record.get(key)
+    if not isinstance(value, str) or not value.strip():
+        raise PolicyError(f"{label} has missing/invalid {key}")
+    return value
+
+
+def validate_and_generate(policy: dict, repo_root: Path) -> list[dict]:
```

Ruff (0.16.0)

[warning] Too many branches (29 > 12)

(PLR0912)